ui/hooks/gasFeeInput/useTransactionFunctions.js

```diff
@@ -87,7 +89,18 @@ export const useTransactionFunctions = ({
         );
         dispatch(updateCustomSwapsEIP1559GasParams(newGasSettings));
       } else {
-        dispatch(updateTransactionFn(updatedTxMeta));
+        const userSettings = {};
+        userSettings.userEditedGasLimit = updatedTxMeta.userEditedGasLimit;
+        userSettings.userFeeLevel = updatedTxMeta.userFeeLevel;
+
+        if (txMeta && txMeta.previousGas) {
+          dispatch(
```

Comment:
We should await this

Reply:
Fixed [here](https://github.com/MetaMask/metamask-extension/pull/13646/commits/1669aac0914ec5681b651a1e3a76546f5108bd3e)